### DISCUSSION TITLE
What is the default populate depth rule?

### DISCUSSION THREAD

**Discussion by yagotome** [strapi/strapi#6242] [Legacy Topics]
I realized that nested relationships aren't auto populated by default on `find`. However, I haven't found anything in docs about this rule. Furthermore, there are some types (such as files) that are auto populated even in nested objects. So how's the rule?

For example, consider this pseudo result from GET `/my-objects`: 

```jsonc
[
    {
        // depth = 1
        "id": 1,
        "name": "foo",
        "myRelatedObject": {
            // depth = 2
            "id": 20,
            "name": "bar",

            // depth = 3; this object is not populated by default, its id is returned instead
            "myNestedRelatedObject": 300,

            "myNestedRelatedFile": {
                // depth = 3; this is populated by default
                "id": 99,
                "mime": "image/png"
                ...
            }
        },
        "created_at": "2020-05-17T18:07:59.366Z",
        "updated_at": "2020-05-17T23:08:53.765Z"
    }
]
```

Is 3 the default depth? How it works? Could it be increases somehow?

I'd like to have objects with depth equal to 3 being populated automatically. Is it possible to change this behavior with some param?

**Comment by lauriejim** [comment]
Hello!
By default we will have level 1 and 2 by default.
In your case `myNestedRelatedFile` is an image and we populate media data in any case.
If you want to manage that, you will have to customize the `populate` params of your API.
Here is a video that shows you how to do - https://youtu.be/VBNjCgUokLk

**Reply by gtamas** [reply]
@lauriejim 
@soupette
@derrickmehaffy  
@anyone :)

I'm using strapi 3.0.5 and it's not working. I cannot populate more than one level.

I have this relation in xy.model.settings.json

```
 "master_content": {
      "model": "master-content",
      "via": "contents"
    }
```

master-content has more relations:

```
 "contentCategory": {
      "model": "content-category"
    },
    "tenant": {
      "model": "tenant"
    }
```

I'm trying to populate like this:

await strapi.query('model1').find(someQuery, ['master_content', 'master_content.tenant'])

Only the first level gets populated... How dow I fix this?

**Comment by gtamas** [comment]
@yagotome 
Does this work for you?
